Sources/LLBuildManifest/BuildManifest.swift

```diff
@@ -87,6 +114,16 @@ public struct BuildManifest {
         commands[name] = Command(name: name, tool: tool)
     }
 
+    public mutating func addWriteLinkFileListCommand(
+        objects: [AbsolutePath],
+        linkFileListPath: AbsolutePath
+    ) {
+        let inputs = [.virtual(WriteAuxiliary.LinkFileList.name)] + objects.map { Node.file($0) }
```

Comment:
so the firs entry is the name of the file and `outputFilePath` is the directory?

Reply:
There's no real API here, only what we can store in either the build description or the LLBuild manifest. I kind of don't want to introduce many more additional types into the build description, so we're limited by the current capabilities which is storing inputs and outputs only.

Reply:
Going to leave a FIXME here to improve in the future, especially helpful would be more clients for a more bespoke API to not shape it entirely around a single use case. 

I'm going to make a small improvement and move this code into `WriteAuxiliary.LinkFileList` so that the knowledge about transformation to/from the inputs array is localized in a single type and the rest of the codebase uses a typed API.

Reply:
Updated now to do this and also added `WriteAuxiliary.fileTypes` to avoid having to add to a random switch statement anytime a new kind of auxiliary file would be added. You now just add the type and the entry point for adding the command.